oarepo_dashboard/ui/dashboard_components/semantic-ui/js/dashboard_components/search/FacetsButtonGroupNameToggler.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext } from "react";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/oarepo_dashboard";
+import { Button } from "semantic-ui-react";
+import { withState } from "react-searchkit";
+import { SearchConfigurationContext } from "@js/invenio_search_ui/components";
+
+const FacetsButtonGroupNameTogglerComponent = ({
+  currentResultsState,
+  currentQueryState,
+  updateQueryState,
+  facetNames,
+  firstFacetButtonText,
+  secondFacetButtonText,
+  keepFiltersOnUpdate,
+  ...uiProps
+}) => {
+  const { initialQueryState } = useContext(SearchConfigurationContext);
+  const currentFilter = currentQueryState.filters?.find((f) =>
+    facetNames.includes(f[0])
+  );
+  const initialQueryFacets = initialQueryState.filters?.map((f) => f[0]);
+  if (!currentFilter)
+    console.error(
+      "FacetsButtonGroup: Query does not contain any of the facets you wish to toggle between, please make sure you are passing initialFilters properly"
+    );
+  const facetStatus = currentFilter && JSON.parse(currentFilter?.[1]);
+  const handleFacetNameChange = (facetName) => {
+    if (currentFilter[0] === facetName) return;
+
+    currentQueryState.filters = keepFiltersOnUpdate
```

Comment:
I've seen similar logic in many places already, it might call for extraction into some reusable utility function

Reply:
@mirekys There is a similar function in the group that toggles facet value and group that toggles facet name, but it is not quite the same and I don't see that it can be extracted to a utility function personally, without making the function excessivelly complicated.